src/services/squad_spawner/squad_spawner.cpp

```diff
@@ -0,0 +1,301 @@
+
+			//Catch position change of Disperse and revert
+			s.m_spawn_pos = original_pos;
+
+			if (entity::take_control_of(s.m_members[i].handle))
+			{
+				if (veh_spawned)
+					PED::SET_PED_INTO_VEHICLE(s.m_members[i].handle, s.m_veh_handle, (i - 1));
+				PED::SET_PED_RELATIONSHIP_GROUP_HASH(s.m_members[i].handle, RAGE_JOAAT("HATES_PLAYER"));
+
+				int seq;
+				TASK::OPEN_SEQUENCE_TASK(&seq);
+				if (veh_spawned)
+				{
+					if (i == 0)
+					{
+						if (VEHICLE::IS_THIS_MODEL_A_CAR(veh_model_hash))
+						{
+							TASK::TASK_VEHICLE_MISSION_PED_TARGET(0, s.m_veh_handle, s.current_target_ped, s.m_stay_in_veh ? 6 : 4, 100.f, 786468, 12.f, 5.f, true);
+							if (!s.m_stay_in_veh)
+								TASK::TASK_LEAVE_ANY_VEHICLE(0, 0, 16);
+						}
+						else if (VEHICLE::IS_THIS_MODEL_A_HELI(veh_model_hash))
+						{
+							TASK::TASK_HELI_MISSION(0, s.m_veh_handle, 0, s.current_target_ped, 0, 0, 0, 6, 200.f, 30.f, -1, 50.f, 20.f, -1, 128 | 4096);
+							VEHICLE::SET_HELI_BLADES_FULL_SPEED(s.m_veh_handle);
+						}
+						else if (VEHICLE::IS_THIS_MODEL_A_PLANE(veh_model_hash))
+						{
+							TASK::TASK_PLANE_MISSION(0, s.m_veh_handle, 0, s.current_target_ped, 0, 0, 0, 6, 300, 35.f, -1, 100.f, 20.f, true);
+							VEHICLE::SET_VEHICLE_FORWARD_SPEED(s.m_veh_handle, 30.f);
+						}
+						else if (VEHICLE::IS_THIS_MODEL_A_BOAT(veh_model_hash))
+						{
+							TASK::TASK_BOAT_MISSION(0, s.m_veh_handle, 0, s.current_target_ped, 0, 0, 0, 6, 300, 786468, 10.f, 7);
+						}
+					}
+				}
+
+				TASK::TASK_COMBAT_PED(0, s.current_target_ped, 67108864, 16); //flag 67108864 should prevent peds from attaining other targets
+				TASK::SET_SEQUENCE_TO_REPEAT(seq, 1);
+				TASK::CLOSE_SEQUENCE_TASK(seq);
+				TASK::TASK_PERFORM_SEQUENCE(s.m_members[i].handle, seq);
+				TASK::CLEAR_SEQUENCE_TASK(&seq);
+
+				if (s.m_stay_in_veh)
+					PED::SET_PED_COMBAT_ATTRIBUTES(s.m_members[i].handle, 3, false);
+			}
+		}
+
+		if (s.m_spawn_behind_same_velocity && s.does_squad_have_vehicle() && veh_spawned && target_player->get_current_vehicle())
+		{
+			Vehicle target_vehicle = g_pointers->m_gta.m_ptr_to_handle(target_player->get_current_vehicle());
+
+			if (ENTITY::GET_ENTITY_SPEED(target_vehicle) > 25.f && entity::take_control_of(s.m_veh_handle))
+			{
+				Vector3 velocity  = ENTITY::GET_ENTITY_VELOCITY(target_vehicle);
+				Vector3 behindpos = ENTITY::GET_OFFSET_FROM_ENTITY_IN_WORLD_COORDS(target_vehicle, 0.f, -7.f, 0.f);
+				float heading     = ENTITY::GET_ENTITY_HEADING(target_vehicle);
+
+				ENTITY::SET_ENTITY_COORDS(s.m_veh_handle, behindpos.x, behindpos.y, behindpos.z, 0, 0, 0, false);
+				ENTITY::SET_ENTITY_HEADING(s.m_veh_handle, heading);
+				ENTITY::SET_ENTITY_VELOCITY(s.m_veh_handle, velocity.x, velocity.y, velocity.z);
+			}
+		}
+
+
+		if (s.has_squad_spawned())
+		{
+			m_active_squads.push_back(s);
+			return true;
+		}
+		return false;
+	}
+
+	void squad_spawner::tick()
+	{
+		for (auto& s : m_active_squads)
+		{
+			if (s.is_squad_alive())
```

Comment:
Left that there to figure out a way how to check whether the attackers have their targets after the player dies. Ive tried about 4 methods including the SET_SEQUENCE_TO_REPEAT repeat. Im afraid that doesn't always work due to their previous task never finishing. Any idea how I could check if a player ped has respawned and make sure the attackers are tasked accordingly?

Reply:
Thats exactly the method I use for the Vehicle controller. However, it seems ped attacking tasks are varied to the point I that checking the required ones overlap certain cases id like to avoid. Since the feature has vehicle capabilities, the tasks the peds issue for the required order range from the simplest DoNothing to ExitVehicle. I have tried scrutinizing the squad vehicle with the native is mission active but that failed as soon as the peds had to leave the vehicle, since that is paired with the end of said mission. For on-foot attackers, checking with is ped in combat native is enough. I just have to consider all scenarios.